cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -718,6 +719,28 @@ public void liveInfo(HttpRequest request, HttpResponder responder, @PathParam("n
                 runtimeService);
   }
 
+  @GET
+  @Path("/apps/{app-id}/schedules/{program-type}/{program-name}")
+  public void programSchedules(HttpRequest request, HttpResponder responder,
```

Comment:
Is there a test for this API? We should probably re-use/refactor ``ProgramLifecycleHttpHandlerTest.testWorkflowSchedules()`` to test this API. When v3 APIs for schedules were added, ``namespaceId`` was not being propagated for schedules (https://issues.cask.co/browse/CDAP-1073). It would probably be a good idea to test if that has changed with these changes, although I don't think it has. 

Basically, we should have a test that deploys to 1 namespace, and ensures that the schedule is not available in the other namespace. Let's talk about it in person, maybe we can do it in another PR. I brought it up because this PR is changing a lot of the schedule code, so it would be a good idea to think of how we could fix CDAP-1073 given these changes.